prereise/gather/griddata/transmission/geometry.py

```diff
@@ -0,0 +1,272 @@
+                "ac": _euclidian(self.a, (self.c[0], -self.c[1])),
+                "bc": _euclidian(self.b, (self.c[0], -self.c[1])),
+            }
+            # 'Equivalent' distances are geometric means
+            self.equivalent_reflected_distance = _geometric_mean(
+                self.reflected_distance.values()
+            )
+            self.equivalent_height = _geometric_mean([self.a[1], self.b[1], self.c[1]])
+        elif self.locations.circuits == 2:
+            # Multi-circuit, so e.g. self.a = ((x0, y0), (x1, y1))
+            # For simplicity, we don't use reflections for multi-circuit capacitance
+            self.true_distance = {
+                "ab": _geometric_mean_euclidian(self.a, self.b),
+                "ac": _geometric_mean_euclidian(self.a, self.c),
+                "bc": _geometric_mean_euclidian(self.b, self.c),
+            }
+            self.phase_self_distances = [
+                _geometric_mean(
+                    [
+                        _euclidian(p0, p1)
+                        for i, p0 in enumerate(phase)
+                        for j, p1 in enumerate(phase)
+                        if i > j
+                    ]
+                )
+                for phase in (self.a, self.b, self.c)
+            ]
+        # 'Equivalent' distances are geometric means
+        self.equivalent_distance = _geometric_mean(self.true_distance.values())
+
+    def calculate_inductance_per_km(self):
+        if self.locations.circuits == 1:
+            denominator = self.bundle.spacing_L
+        else:
+            phase_gmrs = [
+                _geometric_mean([p, self.bundle.spacing_L])
+                for p in self.phase_self_distances
+            ]
+            denominator = _geometric_mean(phase_gmrs)
+        inductance_per_km = (
+            mu_0 / (2 * pi) * log(self.equivalent_distance / denominator)
+        )
+        return inductance_per_km
+
+    def calculate_shunt_capacitance_per_km(self):
+        if self.locations.circuits == 1:
+            capacitance_per_km = (2 * pi * epsilon_0) / (
+                log(self.equivalent_distance / self.bundle.spacing_C)
+                - log(self.equivalent_reflected_distance / (2 * self.equivalent_height))
+            )
+        else:
+            phase_gmrs = [
+                _geometric_mean([p, self.bundle.spacing_C])
+                for p in self.phase_self_distances
+            ]
+            denominator = _geometric_mean(phase_gmrs)
+            capacitance_per_km = (2 * pi * epsilon_0) / (
+                log(self.equivalent_distance / denominator)
+            )
+        return capacitance_per_km
+
+
+def _euclidian(a, b):
+    try:
+        if len(a) != len(b):
+            raise ValueError("Length of a and b must be equivalent")
+    except TypeError:
+        raise TypeError(
+            "a and b must both be iterables compatible with the len() function"
+        )
+    return sqrt(sum((a[i] - b[i]) ** 2 for i in range(len(a))))
```

Comment:
Could do `sqrt(sum((x-y)**2 for x,y in zip(a,b)))`

Reply:
Nice, done.